frontend/internal-packages/agent/src/workflow/nodes/validateInitialSchemaNode.ts

```diff
@@ -1,27 +1,38 @@
+import { randomUUID } from 'node:crypto'
+import { dispatchCustomEvent } from '@langchain/core/callbacks/dispatch'
+import { AIMessage } from '@langchain/core/messages'
 import { executeQuery } from '@liam-hq/pglite-server'
 import { isEmptySchema, postgresqlSchemaDeparser } from '@liam-hq/schema'
+import { SSE_EVENTS } from '../../streaming/constants'
 import type { WorkflowState } from '../../types'
 import { WorkflowTerminationError } from '../../utils/errorHandling'
 
-const createValidationError = (
-  error: Error | string,
-): WorkflowTerminationError => {
-  const errorInstance = error instanceof Error ? error : new Error(error)
-  return new WorkflowTerminationError(
-    errorInstance,
-    'validateInitialSchemaNode',
-  )
-}
-
 /**
  * Validates initial schema and provides Instant Database initialization experience.
  * Only runs on first workflow execution.
  */
+
+async function handleValidationError(
+  userMessage: string,
+  errorMessage: string,
+): Promise<never> {
+  const aiMessage = new AIMessage({
+    id: randomUUID(),
+    content: userMessage,
+  })
+
+  await dispatchCustomEvent(SSE_EVENTS.MESSAGES, aiMessage)
```

Comment:
In this implementation, since the message isn't saved in langgraph's state, I think it will disappear after reloading.

Reply:
@MH4GF 
You're absolutely right. I'm trying to separate the error message display process from the error information retrieval process during reloading.

Should I create a pull request after completing all investigations?🤔

> Next Pull Request
Retrieve error information during reloading
Improve error message display content and design